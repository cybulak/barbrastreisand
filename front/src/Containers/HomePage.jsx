import React from 'react';
import { connect } from 'react-redux';

import { Creators as AuthActions } from '../Reducer/authReducer';

export class HomePage extends React.Component {
	render() {
		return (
			<div>
				<h2>Logged In</h2>
				<button onClick={this.props.logout}>Logout</button>
			</div>
		);
	}
}

const mapDispatchToProps = dispatch => {
	return {
		logout: () => dispatch(AuthActions.logout())
	};
};

export default connect(
	null,
	mapDispatchToProps
)(HomePage);
