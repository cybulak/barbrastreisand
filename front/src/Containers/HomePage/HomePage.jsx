import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Creators as AuthActions } from '../../Reducer/authReducer';

export const HomePage = ({ logout }) => (
	<div>
		<h2>Logged In</h2>
		<button onClick={logout}>Logout</button>
	</div>
);

const mapDispatchToProps = dispatch => ({
	logout: () => dispatch(AuthActions.logout())
});

HomePage.propTypes = {
	logout: PropTypes.func
};

export default connect(
	null,
	mapDispatchToProps
)(HomePage);
