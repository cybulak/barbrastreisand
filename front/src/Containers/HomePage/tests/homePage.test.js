import React from 'react';
import renderer from 'react-test-renderer';
import { HomePage } from '../HomePage';

test('HomePage renders correctly', () => {
	let tree = renderer.create(<HomePage />).toJSON();
	expect(tree).toMatchSnapshot();
});
