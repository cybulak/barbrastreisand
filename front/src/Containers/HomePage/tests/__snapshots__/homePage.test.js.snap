// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div>
  <h2>
    Logged In
  </h2>
  <button>
    Logout
  </button>
</div>
`;
